packages/plugin-page-view-tracking/test/page-view-tracking.test.ts

```diff
@@ -0,0 +1,311 @@
+import { createInstance } from '@amplitude/analytics-browser';
+import { PageViewTrackingPlugin } from '../src/page-view-tracking';
+import { PluginCampaignTracker } from '../src/plugin-campaign-tracker';
+
+describe('PageViewTrackingPlugin', () => {
+  beforeAll(() => {
+    Object.defineProperty(window, 'location', {
+      value: {
+        hostname: '',
+        href: '',
+        pathname: '',
+        search: '',
+        writable: true,
+      },
+    });
+  });
+
+  afterAll(() => {
+    Object.defineProperty(window, 'location', {
+      value: {
+        hostname: '',
+        href: '',
+        pathname: '',
+        search: '',
+        writable: false,
+      },
+    });
+  });
+
+  const API_KEY = 'API_KEY';
+  const USER_ID = 'USER_ID';
+
+  describe('setup', () => {
+    describe('should send a page view event', () => {
+      test('when a campaign changes and filter is "onAttribution"', async () => {
+        const instance = createInstance();
+        const track = jest.spyOn(instance, 'track').mockReturnValueOnce({
+          promise: Promise.resolve({
+            code: 200,
+            message: '',
+            event: {
+              event_type: 'event_type',
+            },
+          }),
+        });
```

Comment:
Maybe do this in beforeEach?

Reply:
perhaps better for making it DRY. i'd prefer that spies and mocks are in test itself since they should be custom to the test case.